Classes/common/Encryption/Keychain/CDTEncryptionKeychainUtils.m

```diff
@@ -0,0 +1,151 @@
+//
+//  CDTEncryptionKeychainUtils.m
+//
+//
+//  Created by Enrique de la Torre Fernandez on 09/04/2015.
+//
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import "CDTEncryptionKeychainUtils.h"
+#import "CDTEncryptionKeychainUtils+AES.h"
+#import "CDTEncryptionKeychainUtils+PBKDF2.h"
+
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_KEYGEN_LABEL = @"KEYGEN_ERROR";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_KEYGEN_MSG_INVALID_ITERATIONS =
+    @"Number of iterations must greater than 0";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_KEYGEN_MSG_INVALID_LENGTH =
+    @"Length must greater than 0";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_KEYGEN_MSG_EMPTY_PASSWORD =
+    @"Password cannot be nil/empty";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_KEYGEN_MSG_EMPTY_SALT =
+    @"Salt cannot be nil/empty";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_KEYGEN_MSG_PASS_NOT_DERIVED =
+    @"Password not derived";
+
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_ENCRYPT_LABEL = @"ENCRYPT_ERROR";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_ENCRYPT_MSG_EMPTY_DATA =
+    @"Cannot encrypt empty/nil data";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_ENCRYPT_MSG_EMPTY_KEY =
+    @"Cannot work with an empty/nil key";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_ENCRYPT_MSG_EMPTY_IV =
+    @"Cannot encrypt with empty/nil iv";
+
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_DECRYPT_LABEL = @"DECRYPT_ERROR";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_DECRYPT_MSG_EMPTY_CIPHER =
+    @"Cannot decrypt empty/nil cipher";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_DECRYPT_MSG_EMPTY_KEY =
+    @"Cannot work with an empty/nil key";
+NSString *const CDTENCRYPTION_KEYCHAIN_UTILS_ERROR_DECRYPT_MSG_EMPTY_IV =
+    @"Cannot decrypt with empty/nil iv";
+
+@interface CDTEncryptionKeychainUtils ()
+
+@end
+
+@implementation CDTEncryptionKeychainUtils
+
+#pragma mark - Public class methods
++ (NSData *)generateRandomBytesInBufferWithLength:(NSUInteger)length
+{
+    uint8_t randBytes[length];
+    
+    int rc = SecRandomCopyBytes(kSecRandomDefault, (size_t)length, randBytes);
```

Comment:
Like this:

```objc
NSAssert((size_t)length <= SIZE_T_MAX, @"length %lu out of bound", (unsigned long)length);
```

This method is only used by `CDTEncryptionKeychainManager`, I do not think we need a more complicate validation.